es-ds-components/components/es-toggle.vue

```diff
@@ -0,0 +1,133 @@
+<script setup lang="ts">
+import { SwitchRoot, SwitchThumb } from 'reka-ui';
+
+const switchId = useId();
+const model = defineModel<boolean>({
+    default: false,
+});
+
+defineProps({
+    disabled: {
+        type: Boolean,
+        default: false,
+        description: 'Disables the toggle.',
+    },
+});
+
+// Workaround: Reka UI components don't play nice with the Nuxt 3.16+ dev server side rendering while using NPM link
+const wrappingComponent = import.meta.dev ? resolveComponent('ClientOnly') : 'span';
+</script>
+
+<template>
+    <component :is="wrappingComponent">
+        <div
+            class="d-flex flex-row"
+            :class="{
+                'has-label': $slots.label,
+            }">
+            <switch-root
+                :id="switchId"
+                v-model="model"
+                :disabled="disabled"
+                class="es-toggle">
+                <switch-thumb class="es-toggle-thumb" />
+            </switch-root>
+            <label
+                v-if="$slots.label"
+                :for="switchId"
+                class="es-toggle-label">
+                <slot name="label" />
+            </label>
+        </div>
+    </component>
+</template>
+
+<style lang="scss">
+@use '@energysage/es-ds-styles/scss/variables' as variables;
+
+.has-label {
+    gap: 16px;
+    align-items: center;
+
+    .es-toggle[data-disabled] ~ .es-toggle-label {
+        opacity: 0.5;
+
+        :deep(*) {
+            opacity: inherit;
+        }
+    }
+}
+
+.es-toggle-label {
+    margin-top: 8px;
```

Comment:
Worth throwing a `user-select: none` in here so this text highlighting doesn't happen if you click too quickly?

https://github.com/user-attachments/assets/b918944a-2491-4065-a28d-db52e351b944